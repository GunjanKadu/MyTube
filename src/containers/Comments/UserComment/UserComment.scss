.comment {
    display: flex;
    margin: 8px 0px;

    & > div {
        /*
            Prevent child from becoming bigger than parent by setting the min-width to zero
            The div might overflow if there is an extremely long word without spaces
            reference: https://bugzilla.mozilla.org/show_bug.cgi?id=1108514
        */
        min-width: 0px;
        span {
            overflow-wrap: break-word;
        }
    }

    .user-image {
        width: 48px;
        height: 48px;
        margin-right: 10px;
        flex-grow: 0;
        flex-shrink: 0;
    }

    .user-name {
        font-weight: 600;
        margin-bottom: 4px;
    }
    .comment-actions {
        margin-top: 4px;
    }
}



