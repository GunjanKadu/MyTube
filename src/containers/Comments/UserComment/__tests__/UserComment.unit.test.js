import React from 'react';
import {shallow} from 'enzyme';
import {UserComment} from '../UserComment';

const comment = {
  "kind": "youtube#commentThread",
  "etag": "\"XI7nbFXulYBIpL0ayR_gDh3eu1k/7qOYhKtSTLDez0FUuT8Axyuyhvg\"",
  "id": "UgycuaK9UFzPyBHmgNJ4AaABAg",
  "snippet": {
    "videoId": "OVazJLD5lM8",
    "topLevelComment": {
      "kind": "youtube#comment",
      "etag": "\"XI7nbFXulYBIpL0ayR_gDh3eu1k/iPL4MvxUFXz1L8Rgn1EAXvIVU5s\"",
      "id": "UgycuaK9UFzPyBHmgNJ4AaABAg",
      "snippet": {
        "authorDisplayName": "Peter Unlustig",
        "authorProfileImageUrl": "https://yt3.ggpht.com/-QO3910-PfI0/AAAAAAAAAAI/AAAAAAAAAAA/XiVbzk5N7TI/s28-c-k-no-mo-rj-c0xffffff/photo.jpg",
        "authorChannelUrl": "http://www.youtube.com/channel/UC5hCkka0HvWJ0PqRFhUDaig",
        "authorChannelId": {
          "value": "UC5hCkka0HvWJ0PqRFhUDaig"
        },
        "videoId": "OVazJLD5lM8",
        "textDisplay": "WTF is this setup … what does he need 20 channels for?",
        "textOriginal": "WTF is this setup … what does he need 20 channels for?",
        "canRate": true,
        "viewerRating": "none",
        "likeCount": 0,
        "publishedAt": "2018-07-27T07:42:12.000Z",
        "updatedAt": "2018-07-27T07:42:12.000Z"
      }
    },
    "canReply": true,
    "totalReplyCount": 0,
    "isPublic": true
  }
};

test('renders UserComment with props.comment = null', () => {
  const wrapper = shallow(<UserComment/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders UserComment with valid comment', () => {
  const wrapper = shallow(<UserComment comment={comment}/>);
  expect(wrapper).toMatchSnapshot();
});