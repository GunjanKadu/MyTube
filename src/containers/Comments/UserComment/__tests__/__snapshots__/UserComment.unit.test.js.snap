// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserComment with props.comment = null 1`] = `<div />`;

exports[`renders UserComment with valid comment 1`] = `
<div
  className="comment"
>
  <Image
    as="img"
    circular={true}
    className="user-image"
    src="https://yt3.ggpht.com/-QO3910-PfI0/AAAAAAAAAAI/AAAAAAAAAAA/XiVbzk5N7TI/s28-c-k-no-mo-rj-c0xffffff/photo.jpg"
    ui={true}
  />
  <div>
    <div
      className="user-name"
    >
      Peter Unlustig
    </div>
    <span>
      WTF is this setup â€¦ what does he need 20 channels for?
    </span>
    <div
      className="comment-actions"
    >
      <Rating
        likeCount={0}
      />
       
      <Button
        as="button"
        compact={true}
        role="button"
        size="mini"
      >
        REPLY
      </Button>
    </div>
  </div>
</div>
`;
